mythril/analysis/solver.py

```diff
@@ -122,12 +124,52 @@ def get_transaction_sequence(
         ).as_long()
 
     concrete_initial_state = _get_concrete_state(initial_accounts, min_price_dict)
-
+    if isinstance(transaction_sequence[0], ContractCreationTransaction):
+        _replace_with_actual_sha(concrete_transactions[1:], model)
+    else:
+        _replace_with_actual_sha(concrete_transactions, model)
     steps = {"initialState": concrete_initial_state, "steps": concrete_transactions}
 
     return steps
 
 
+def _replace_with_actual_sha(
+    concrete_transactions: List[Dict[str, str]], model: z3.Model
+):
+    for tx in concrete_transactions:
+        if hash_matcher not in tx["input"]:
+            continue
+        for i in range(0, len(tx["input"]) - 73, 64):
+            if hash_matcher not in tx["input"][i + 10 : i + 74]:
+                continue
+            find_input = symbol_factory.BitVecVal(
+                int(tx["input"][10 + i : 74 + i], 16), 256
+            )
+            input_ = None
+            for size in (160, 256, 512):
+                _, inverse = keccak_function_manager.get_function(size)
+                try:
+                    input_ = symbol_factory.BitVecVal(
+                        model.eval(inverse(find_input).raw).as_long(), size
+                    )
+                except AttributeError:
+                    continue
+                hex_input = hex(input_.value)[2:]
+                found = False
+                for new_tx in concrete_transactions:
+                    if hex_input in new_tx["input"]:
+                        found = True
+                        break
+                if found:
+                    break
+            if input_ is None:
+                continue
+            keccak = keccak_function_manager.find_keccak(input_)
+            tx["input"] = tx["input"].replace(
+                tx["input"][10 + i : 74 + i], hex(keccak.value)[2:]
```

Comment:
calldata is assumed as some sets of 64 bytes from solidity compilation perspective, so it will already be 0 padded I think.I have seen lot's of cases where addresses and uint8's are automatically zero padded by the compiler. Will add a warning when things go shorter than 64 bytes, probably when solidity guys have a change in mindset and can detect it.

Reply:
Nevermind, it's the output right, so we can just ignore them as they aren't hashes.